packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -263,14 +265,14 @@ export class SummarizerNode implements IRootSummarizerNode {
             }
         }
 
-        const snapshotTree = await getSnapshot();
-        const referenceSequenceNumber = await seqFromTree(snapshotTree, readAndParseBlob);
-
-        // If we have seen a summary same or later as the downloaded one, ignore it.
-        if (this.referenceSequenceNumber >= referenceSequenceNumber) {
+        // If we have seen a summary same or later as the current one, ignore it.
+        if (this.referenceSequenceNumber >= summaryRefSeq) {
             return { latestSummaryUpdated: false };
         }
 
+        const snapshotTree = await getSnapshot();
+        const referenceSequenceNumber = await seqFromTree(snapshotTree, readAndParseBlob);
+
         await this.refreshLatestSummaryFromSnapshot(
             referenceSequenceNumber,
```

Comment:
`summaryRefSeq` should be passed directly here. There is no need to extract the `referenceSequenceNumber` from the snapshot.